src/dotty/tools/dotc/core/pickling/TreePickler.scala

```diff
@@ -531,8 +535,16 @@ class TreePickler(pickler: TastyPickler) {
       withLength { pickleType(ann.symbol.typeRef); pickleTree(ann.tree) }
     }
 
+    def updateMapWithDeltas[T](mp: collection.mutable.Map[T, Addr]) = {
+      mp.map{
+        case (key, addr) => (key, adjusted(addr))
```

Comment:
That's way I wrote the code this way: at the point when I am updating the datastructure, I'm already iterating over a different one.